### PR TITLE
Handle blank entries in homophones.csv

### PR DESCRIPTION
**Pull request by geobarry**
added a line to function update_homophones in homophones.py to remove words that are empty strings from the list of homophones created from each line of the CSV file. The issue is that CSV file format is not standard, and programs like MS Excel try to force each line to have the same number of "cells" (i.e. commas). This results in almost every line having on extra comma which is read as a "word" that is an empty string, and then these empty string "homophones" are matched with other words' empty string homophones resulting in every word being a homophone of every other word in the file.

### CODE REVIEW

File: core/homophones/homophones.py [hunk 1]
```diff
@@ -118,6 +119,7 @@ def raise_homophones(word_to_find_homophones_for, forced=False, selection=False)
         return
 
     # Move current word to end of list to reduce searcher's cognitive load
+
     valid_homophones_reordered = list(
```
Comment (phillco) [inline core/homophones/homophones.py:123]:
```suggestion
    valid_homophones_reordered = list(
```

File: core/homophones/homophones.py [hunk 2]
```diff
@@ -133,14 +135,14 @@ def raise_homophones(word_to_find_homophones_for, forced=False, selection=False)
         and quick_replace
         and not force_raise
     ):
+
         if word_to_find_homophones_for == active_word_list[0].lower():
```
Comment (phillco) [inline core/homophones/homophones.py:139]:
```suggestion
        if word_to_find_homophones_for == active_word_list[0].lower():
```

**Review by nriley — APPROVED** [review]

File: core/homophones/homophones.py [hunk 3]
```diff
@@ -102,6 +102,7 @@ def raise_homophones(word_to_find_homophones_for, forced=False, selection=False)
 
     # We support plurals, but very naively. If we can't find your word but your word ends in an s, presume its plural
     # and attempt to find the singular, then present the presumed plurals back. This could be improved!
+
     if word_to_find_homophones_for in all_homophones:
         valid_homophones = all_homophones[word_to_find_homophones_for]
```
Comment (phillco) [inline core/homophones/homophones.py:107]:
```suggestion
    if word_to_find_homophones_for in all_homophones:
        valid_homophones = all_homophones[word_to_find_homophones_for]
```

**Comment by phillco** [comment]
(Just removing the extra blank lines)

**Pull request merged**